raiden/transfer/channel.py

```diff
@@ -814,6 +831,114 @@ def is_valid_unlock(
     return result
 
 
+def is_valid_action_withdraw(
+    channel_state: NettingChannelState, withdraw: ActionChannelWithdraw
+) -> SuccessOrError:
+    result: SuccessOrError
+
+    balance = get_balance(sender=channel_state.our_state, receiver=channel_state.partner_state)
+
+    withdraw_amount = withdraw.total_withdraw - channel_state.our_state.total_withdraw
+    if withdraw_amount <= 0:
+        msg = "Total withdraw {} did not increase".format(withdraw.total_withdraw)
+        result = (False, msg)
+    elif balance < withdraw_amount:
+        msg = "Insufficient balance: {}. Requested {} for withdraw".format(
+            balance, withdraw_amount
+        )
+        result = (False, msg)
+    else:
+        result = (True, None)
+
+    return result
+
+
+def is_valid_withdraw_request(
```

Comment:
Please add the checks for the chain_id, token_address, channel_id, and for overflows.